common/src/lib.rs

```diff
@@ -644,6 +644,7 @@ pub type NodeIdContent = NonZeroU128;
 
 /// The stable identity of a [`Node`], unique within the node's tree.
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[repr(transparent)]
```

Comment:
Currently, on all types that have a `#repr(...)] attribute, it comes last, just before the actual struct declaration. I'd suggest moving it after the `#cfg|...]` attributes to be consistent.